sdk/daml-lf/transaction/src/test/scala/com/digitalasset/daml/lf/crypto/NodeHashSpec.scala

```diff
@@ -0,0 +1,1028 @@
+                                      |'00000004' # 4 (int)
+                                      |'6e616d65' # name (string)
+                                      |# Signatories
+                                      |'00000001' # 1 (int)
+                                      |'00000005' # 5 (int)
+                                      |'616c696365' # alice (string)
+                                      |# Stakeholders
+                                      |'00000001' # 1 (int)
+                                      |'00000007' # 7 (int)
+                                      |'636861726c6965' # charlie (string)
+                                      |# Acting Parties
+                                      |'00000002' # 2 (int)
+                                      |'00000005' # 5 (int)
+                                      |'616c696365' # alice (string)
+                                      |'00000003' # 3 (int)
+                                      |'626f62' # bob (string)
+                                      |# Interface Id
+                                      |'01' # Some
+                                      |'00000007' # 7 (int)
+                                      |'7061636b616765' # package (string)
+                                      |'00000001' # 1 (int)
+                                      |'00000010' # 16 (int)
+                                      |'696e746572666163655f6d6f64756c65' # interface_module (string)
+                                      |'00000001' # 1 (int)
+                                      |'0000000e' # 14 (int)
+                                      |'696e746572666163655f6e616d65' # interface_name (string)
+                                      |# Choice Id
+                                      |'00000006' # 6 (int)
+                                      |'63686f696365' # choice (string)
+                                      |# Chosen Value
+                                      |'0000000000007a94' # 31380 (long)
+                                      |# Consuming
+                                      |'01' # true (bool)
+                                      |# Exercise Result
+                                      |'01' # Some
+                                      |'00000006' # 6 (int)
+                                      |'726573756c74' # result (string)
+                                      |# Choice Observers
+                                      |'00000001' # 1 (int)
+                                      |'00000005' # 5 (int)
+                                      |'6461766964' # david (string)
+                                      |# Children
+                                      |'00000002' # 2 (int)
+                                      |'$createNodeHash' # (Hashed Inner Node)
+                                      |'$fetchNodeHash' # (Hashed Inner Node)
+                                      |""".stripMargin
+
+      assertStringTracer(hashTracer, hash)
```

Comment:
I would like a more involved transaction, at least with a root node before the exercise and one after.